replica/table.cc

```diff
@@ -609,26 +611,35 @@ compaction_group* table::single_compaction_group_if_available() const noexcept {
     return _compaction_groups.size() == 1 ? &*_compaction_groups[0] : nullptr;
 }
 
-compaction_group& table::compaction_group_for_token(dht::token token) const noexcept {
+size_t table::compaction_group_id_for_token(dht::token token) const noexcept {
     auto idx = _cg_manager->compaction_group_of(token);
     if (idx >= _compaction_groups.size()) {
         on_fatal_internal_error(tlogger, format("compaction_group_for_token: index out of range: idx={} size_log2={} size={} token={}",
                                                 idx, _cg_manager->log2_compaction_groups(), _compaction_groups.size(), token));
     }
-    auto& ret = *_compaction_groups[idx];
-    if (!ret.token_range().contains(token, dht::token_comparator())) {
+    return idx;
+}
+
+compaction_group& table::compaction_group_for_token(dht::token token) const noexcept {
+    auto idx = compaction_group_id_for_token(token);
+    auto& cg_ptr = _compaction_groups[idx];
+    if (!cg_ptr) {
+        on_fatal_internal_error(tlogger, format("compaction group idx={}, that owns token={}, is not allocated on this shard",
```

Comment:
I think aborting the node is too harsh, can't we just throw?

Reply:
Coordinators are protected by barriers executed on topology changes. Topology change should drain old requests before moving the tablet away.  There's also fencing which rejects old requests on admission to replica side.

It's good not to assume too much if we don't have to.